BHoM/Dynamo_Toolkit
# Remove legacy code and compilation warnings

### Issues addressed by this PR


Closes #180 




### Test files
- Make sure code compiles without warnings
- Try a few of your existing scripts to make sure everything is still working
- Create a CustomObject component and copy/paste it to make sure serialisation works fine. ~~You will require to have the BHoM on [this pending PR](https://github.com/BHoM/BHoM_UI/pull/160) for everything to work fine but this PR doesn't depend on it (Just want to make sure you don't report bugs that are already taken care of 😄 )~~
